text/0040-token-metadata.md

```diff
@@ -0,0 +1,167 @@
+to the user.
+
+One way to do this is to hard-code all metadata into clients, which might work initially. After this however, this will mean that new tokens
+that are created on chain can't actually be used by users until a client update has been released and the users have updated.
+
+(In the [MCIP #31](https://github.com/mobilecoinfoundation/mcips/pull/0025) implementation PR, we did add a proto enum called `KnownTokenIds`
+which can be shipped with clients, as a stop-gap measure.)
+
+If the clients can dynamically query this information, then a client update is not required for clients to make use of new tokens which are
+created, which significantly reduces the amount of time it takes for a new token to be available to users.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Token metadata includes the following:
+
+* Short (exchange ticker) name of a token, e.g. MOB. (This is not expected to be translated.)
+* Long (english) name of a token, e.g. MobileCoin. (This is expected to be translated, but the client must maintain the translation list.)
+* Display conversion factor, e.g. 10^12. This conversion factor converts from the smallest representable units (`u64`) which appear in the blockchain,
+  to the user-displayable unit. (Note that this is possibly not a power of 10.) `u64` amounts
+  which are obtained from `TxOut`'s on chain should be divided by this number, for the token id, before being displayed to the user.
+* (Optional) URL pointing to an SVG logo of the token. (This is expected to be scalable and appropriate for display on devices of many form factors.)
+* (Optional) URL pointing to an MCIP which proposed the token.
+
+The consensus network will provide an endpoint at which this metadata can be requested for one or more token ids.
+
+This data will be provided at startup time via a file to the consensus node. (In the future, we may consider to store this data on chain.)
+The node should require this data for any token which is configured to exist (has a minimum fee in `tokens.toml`).
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+A token metadata proto has these fields
+
+```
+message TokenMetadata {
+    // The short name is a standard ticker symbol for the token
+    String short_name = 1;
+    // The long name is the standard english name for the token
+    String long_name = 2;
+    // The conversion factor is an unsigned 64 bit number expressed as a decimal string
+    String display_conversion_factor = 3;
+    // If present, a url pointing to a logo for the token
+    String logo_url = 4;
+    // If present, a url pointing to an MCIP which proposed the token
+    String mcip_url = 5;
+}
+```
+
+A consensus network client-facing API will be exposed to query for these objects given one or more token ids.
+
+The `tokens.toml` file schema is extended to contain these fields per token. The minting trust root should sign
```

Comment:
Right now the `minting trust root` signs the governors and not the minimum fee, the only two configurable options per token. Maybe we should change the configuration to have two distinct sections - signed and unsigned, to make it clear what is under the signature and what isn't.
We can then add the signature to the responder id to ensure all signed configuration on all nodes is identical, which addresses one of the drawbacks mentioned below.

Reply:
maybe it's okay for the governers to sign this instead of the minting trust root?

Reply:
In which case it will likely need to go `MintConfigTx`... or it will be a pretty convoluted process of back and forth between the governors and then the node operators to update the new tokens.toml